File: src/codes/Users_Manual/link/soem_0.cpp
//~#include <iostream>
#include <autd3_link_soem.hpp>

//~int main() {
//~using namespace autd3;
link::SOEM(
    [](const uint16_t slave, const link::Status status) {
      std::cout << "slave [" << slave << "]: " << status << std::endl;
      if (status == link::Status::Lost()) {
        exit(-1);
      }
    },
    link::SOEMOption{.buf_size = 16,
                     .ifname = "",
                     .state_check_interval = std::chrono::milliseconds(100),
                     .sync0_cycle = std::chrono::milliseconds(1),
                     .send_cycle = std::chrono::milliseconds(1),
                     .thread_priority = link::ThreadPriority::Max(),
                     .process_priority = link::ProcessPriority::High,
                     .sync_tolerance = std::chrono::microseconds(1),
                     .sync_timeout = std::chrono::seconds(10),
                     .affinity = std::nullopt

    });
//~return 0; }

File: src/codes/Users_Manual/link/soem_0.rs
use autd3_link_soem::{Status, SOEM, SOEMOption};
# use std::time::Duration;

# fn main() {
# let _ = 
SOEM::new(
    |slave, status| {
        eprintln!("slave [{}]: {}", slave, status);
        if status == Status::Lost {
            std::process::exit(-1);
        }
    },
    SOEMOption {
        ifname: None,
        state_check_interval: Duration::from_millis(100),
        sync0_cycle: Duration::from_millis(1),
        sync_tolerance: Duration::from_micros(1),
        sync_timeout: Duration::from_secs(10),
    },
);
# }